### PR TITLE
Fix library.lookup with an empty uri list

### LINKED ISSUE #1619: Traceback from library.lookup when passing empty list of uris
Whilst fixing up the tests for another PR I noticed we do not properly handle calling `LibraryController.lookup()` with an empty list:

```
$ curl -X POST -H Content-Type:application/json -d '{ "method": "core.library.lookup", "jsonrpc": "2.0", "params": { "uris": [] },  "id": 1 }' http://localhost:6680/mopidy/rpc | json_pp

{
   "jsonrpc" : "2.0",
   "id" : 1,
   "error" : {
      "message" : "Invalid params",
      "code" : -32602,
      "data" : {
         "type" : "TypeError",
         "message" : "'NoneType' object is not iterable",
         "traceback" : "Traceback (most recent call last):\n  File \"/home/nsteel/.virtualenvs/mopidy/local/lib/python2.7/site-packages/mopidy/internal/jsonrpc.py\", line 133, in _handle_single_request\n    result = self._unwrap_result(result)\n  File \"/home/nsteel/.virtualenvs/mopidy/local/lib/python2.7/site-packages/mopidy/internal/jsonrpc.py\", line 222, in _unwrap_result\n    result = result.get()\n  File \"/work/nsteel/projects/mopidy-dev/pykka/pykka/threading.py\", line 52, in get\n    compat.reraise(*self._data['exc_info'])\n  File \"/work/nsteel/projects/mopidy-dev/pykka/pykka/compat.py\", line 12, in reraise\n    exec('raise tp, value, tb')\n  File \"/work/nsteel/projects/mopidy-dev/pykka/pykka/actor.py\", line 201, in _actor_loop\n    response = self._handle_receive(message)\n  File \"/work/nsteel/projects/mopidy-dev/pykka/pykka/actor.py\", line 295, in _handle_receive\n    return callee(*message['args'], **message['kwargs'])\n  File \"/home/nsteel/.virtualenvs/mopidy/local/lib/python2.7/site-packages/mopidy/core/library.py\", line 231, in lookup\n    for u in backend_uris:\nTypeError: 'NoneType' object is not iterable\n"
      }
   }
}
```
* `LibraryController.get_images(uris=[])` is fine, returns an empty dictionary.
* `LibraryController.search(uris=[])` is also fine, it passes the None type straight to the backend instances which are sure to handle that as it's the parameter's default value.

To fix we just need something similar to what `search` already has.

### PR DESCRIPTION
**Pull request by kingosticks**
Fixes #1619.

### CODE REVIEW

**Comment by kingosticks** [comment]
Any comments or shall I just merge?

File: mopidy/core/library.py [hunk 1]
```diff
@@ -227,9 +227,11 @@ def lookup(self, uri=None, uris=None):
         results = {u: [] for u in uris}
 
         # TODO: lookup(uris) to backend APIs
-        for backend, backend_uris in self._get_backends_to_uris(uris).items():
-            for u in backend_uris:
-                futures[(backend, u)] = backend.library.lookup(u)
+        futures = {
```
Comment (adamcik) [inline mopidy/core/library.py]: I find this a lot harder to read and parse with the multiple for loops in this expansion. To the point of it being hard to follow what the real change is.

Comment (kingosticks) [inline mopidy/core/library.py]: I actually totally agree, I pretty much copied it from the other similar place it was used.

**Comment by kingosticks** [comment]
I think this is much clearer.

File: mopidy/core/library.py [hunk 2]
```diff
@@ -228,8 +228,9 @@ def lookup(self, uri=None, uris=None):
 
         # TODO: lookup(uris) to backend APIs
         for backend, backend_uris in self._get_backends_to_uris(uris).items():
-            for u in backend_uris:
-                futures[(backend, u)] = backend.library.lookup(u)
+            if backend_uris:
```
Comment (adamcik) [inline mopidy/core/library.py]: Could also be `for u in backend_uris or []` I think, but either way is fine with me.

Comment (kingosticks) [inline mopidy/core/library.py]: Hehe, Yeh, but I've gone from super obfuscated to super obvious here - no middle ground!
